logstreams/src/main/java/io/zeebe/logstreams/impl/log/LogStreamImpl.java

```diff
@@ -156,12 +193,55 @@ private void internalSetCommitPosition(final long commitPosition) {
 
   @Override
   public ActorFuture<LogStreamRecordWriter> newLogStreamRecordWriter() {
-    return actor.call(() -> new LogStreamWriterImpl(partitionId, writeBuffer));
+    final var writerFuture = new CompletableActorFuture<LogStreamRecordWriter>();
+    actor.call(() -> createWriter(writerFuture, LogStreamWriterImpl::new));
```

Comment:
If using `call`, couldn't `createWriter` not just return the `LogStreamWriter`? Then you don't need to create an extra future. Or just switch to run; these `call` calls (:sweat_smile:) are strange to me when they return nothing.

Reply:
I need to handle this future in the createWriter method. I can change this to `#run` yes.